toolkit/tools/imagegen/configuration/networkconfig.go

```diff
@@ -0,0 +1,309 @@
+	// by dashes instead of colons. Therefore, we're reading from the 10th spot of the
+	// string to obtain the MAC address and then replace the dashes with colons
+	deviceAddr = strings.Replace(bootifValue[macAddressStartIndex:len(bootifValue)], "-", ":", -1)
+	return
+}
+
+func checkNetworkDeviceAvailability(networkData Network) (deviceName string, err error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return
+	}
+
+	if networkData.Device == "bootif" {
+		networkData.Device, err = findBootIfValue()
+		if err != nil {
+			logger.Log.Errorf("Failed to read bootif value from /proc/cmdline")
+			return
+		}
+	}
+
+	for _, iface := range ifaces {
+		if networkData.Device == iface.Name {
+			deviceName = networkData.Device
+			return
+		} else {
+			ifaceAddr := strings.TrimSpace(iface.HardwareAddr.String())
+
+			if strings.Contains(networkData.Device, ":") {
+				if ifaceAddr != "" && strings.EqualFold(ifaceAddr, networkData.Device) {
+					deviceName = iface.Name
+					return
+				}
+			}
+		}
+	}
+
+	return
+}
+
+func populateMatchSection(networkData Network, fileName, deviceName string) (err error) {
+	matchSection := fmt.Sprintf("[Match]\nName=%s\n", deviceName)
+	err = file.Append(matchSection+"\n", fileName)
+	if err != nil {
+		logger.Log.Errorf("Failed to write [Match] section: %s", err)
+	}
+
+	return
+}
+
+func populateNetworkSection(networkData Network, fileName string) (err error) {
+	var networkSection strings.Builder
+
+	// Write network section tag
+	networkSection.WriteString("[Network]\n")
+
+	// Currently only supports static IP setting
+	// Write IP and netmask
+	if networkData.NetMask != "" && networkData.Ip != "" {
+		stringMask := net.IPMask(net.ParseIP(networkData.NetMask).To4())
+		cidrPrefix, _ := stringMask.Size()
+		networkSection.WriteString(fmt.Sprintf("Address=%s/%d\n", networkData.Ip, cidrPrefix))
+	}
+
+	// Write Gateway
+	if networkData.GateWay != "" {
+		networkSection.WriteString(fmt.Sprintf("Gateway=%s\n", networkData.GateWay))
+	}
+
+	// Write Nameserver
+	for _, nameserver := range networkData.NameServers {
+		networkSection.WriteString(fmt.Sprintf("DNS=%s\n", nameserver))
+	}
+
+	err = file.Append(networkSection.String(), fileName)
+	if err != nil {
+		logger.Log.Errorf("Failed to write [Network] section: %s", err)
+	}
+
+	return
+}
+
+func createNetworkConfigFile(installChroot *safechroot.Chroot, networkData Network, deviceName string) (err error) {
+	const (
+		networkFileDir = "/etc/systemd/network"
+		filePrefix     = "10"
+	)
+
+	if exists, ferr := file.DirExists(networkFileDir); ferr != nil {
+		logger.Log.Errorf("Error accessing /etc/systemd/network/")
+		err = ferr
+		return
+	} else if !exists {
+		err = fmt.Errorf("/etc/systemd/network/: no such path or directory")
+		return
+	}
+
+	logger.Log.Debugf("Start creating network file")
+
+	networkFileName := fmt.Sprintf("%s/%s-%s-%s.network", networkFileDir, filePrefix, networkData.BootProto, deviceName)
```

Comment:
>networkFileName [](http://example.com/codeflow?start=1&length=15)

The name is a bit confusing, it's closer to a `networkFilePath`.

Reply:
Fixed